HOWTO.md

```diff
@@ -86,6 +86,66 @@ This will result into this module header:
                                                                    *your_module*
 ```
 
+## Config
+
+You can configure docgen on file basis. For example you can define how `functions` or `classes`
+are sorted.
+
+```lua
+---@config { ['function_order'] = 'ascending', ['class_order'] = 'descending' }
+```
+
+Available keys value pairs are:
+- `function_order`:
+  - `file_order` (default)
+  - `ascending`
+  - `descending`
+  - If you have a typo it will do `file_order` sorting
+- `class_order`:
+  - `file_order` (default)
+  - `ascending`
+  - `descending`
+  - If you have a typo it will do `file_order` sorting
+<!----> TODO(conni2461): DO WE WANT THIS? IF YES IMPLEMENT IT
+<!-- - `field_order`: -->
+<!--   - `file_order` (default) -->
+<!--   - `ascending` -->
+<!--   - `descending` -->
+<!--   - If you have a typo it will do `file_order` sorting -->
+
+## Class
+
+You can define your own classes and types to give a better sense of the Input of another function.
+Another good usecase for this are structs defined by ffi.
+
+Input:
+
+```lua
+---@class Array : Map @number indexed starting at 1
+---@field count number: Always handy to have a count
+---@field type string: Imagine having a type for an array
+---@field begin function: It even has a begin()?! Is this cpp?
+---@field end function: It even has an end()?! Get out of here cpp! Oh by the way did you know that fields are wrapping? I didn't and this should prove this.
+```
+
+Output:
+
+```
+Array : Map                                                            *Array*
```

Comment:
May want to link to a file for these outputs, so they don't go out of date.

Reply:
It would be cool to embed a file content in markdown but if i googled that correctly that not possible, but there is a CI job that does this. Because just linking to a file is not that nice, because people have to open each file.

https://github.com/marketplace/actions/markdown-embed-code-from-file